### PR TITLE
feat(Dropdown): Allow for tooltips on aria-disabled items

### LINKED ISSUE #5826: Support for tooltips on disabled dropdown items (aria-disabled variation)
From a discussion in this issue - https://github.com/patternfly/patternfly-react/issues/5127

We added support for a `.pf-m-aria-disabled` variation on dropdown items to support adding tooltips on disabled items

https://github.com/patternfly/patternfly/pull/4072

cc @jschuler @mcarrano

### PR DESCRIPTION
**Pull request by nicolethoen**
**What**: Closes #5826 

You can see it working in the very first example - one of the disabled items is now using aria-disabled with a simple tooltip.

### CODE REVIEW

**Review by mcarrano — APPROVED** [review]
Looks great @nicolethoen !

**Review by mcoker — APPROVED** [review]
LGTM!

**Review by dlabrecq — DISMISSED** [review]

File: packages/react-core/src/components/Dropdown/InternalDropdownItem.tsx [hunk 1]
```diff
@@ -264,7 +267,7 @@ export class InternalDropdownItem extends React.Component<InternalDropdownItemPr
               role={role}
               onKeyDown={this.onKeyDown}
               onClick={(event: any) => {
-                if (!isDisabled) {
+                if (!isDisabled && !isAriaDisabled) {
```
Comment (kmcfaul) [inline packages/react-core/src/components/Dropdown/InternalDropdownItem.tsx:270]: Should this be an `||` instead? Or does `isDisabled` also set `isAriaDisabled` if `isAriaDisabled` isn't passed in.

**Review by jessiehuff — COMMENTED** [review]
Hmm what's odd is that this seems to work in JAWS, but not in VO and NVDA. I'm not quite sure why yet.

**Comment by nicolethoen** [comment]
@jessiehuff did you do any more investigation into this? I can try playing with it in VO this afternoon.

File: packages/react-core/src/components/Dropdown/InternalDropdownItem.tsx (hunk 1 above)
Comment (tlabaj) [reply packages/react-core/src/components/Dropdown/InternalDropdownItem.tsx:270]: Good question.  Maybe add a test for this scenario.

**Comment by jessiehuff** [comment]
I'm wondering if what Katie pointed out above is affecting the screen reader experience. It seems like if it's just `isAriaDisabled` this might not work on click which the screen reader may rely on? It's also probably worth noting what we did in [button to prevent events](https://github.com/patternfly/patternfly-react/blob/main/packages/react-core/src/components/Button/Button.tsx#L122). Maybe try logging if it gets the onClick and onKeyPress. I would think that the tooltip shouldn't have a problem firing if those are getting triggered.

**Review by tlabaj — COMMENTED** [review]
Can you update the demo app.

File: packages/react-core/src/components/Dropdown/InternalDropdownItem.tsx (hunk 1 above)
Comment (nicolethoen) [reply packages/react-core/src/components/Dropdown/InternalDropdownItem.tsx:270]: I may be going cross eyed - but i think it's true that if the dropdown item is disabled OR ariaDisabled, then we don't want to fire the passed in onClick or onSelect handlers. So this logic, with the `&&` is right.

**Comment by nicolethoen** [comment]
I opened https://github.com/patternfly/patternfly-react/issues/6109 because tooltips on dropdown items are not screenreader accessible.

**Review by tlabaj — APPROVED** [review]

File: packages/react-core/src/components/Dropdown/InternalDropdownItem.tsx (hunk 1 above)
Comment (kmcfaul) [reply packages/react-core/src/components/Dropdown/InternalDropdownItem.tsx:270]: Oh you're right, the double negative with the && got me

**Review by kmcfaul — APPROVED** [review]

File: packages/react-core/src/components/Dropdown/InternalDropdownItem.tsx (hunk 1 above)
Comment (redallen) [reply packages/react-core/src/components/Dropdown/InternalDropdownItem.tsx:270]: Can rewrite as `isDisabled || isAriaDisabled` to be more clear. de Morgan's theorem states `!A && !B = A || B`.

**Review by redallen — APPROVED** [review]

**Pull request merged**